cmd/kube-controller-manager/app/plugins.go

```diff
@@ -65,13 +65,18 @@ import (
 // initialization code for kubelet really, really need a through refactor.
 func ProbeAttachableVolumePlugins() []volume.VolumePlugin {
 	allPlugins := []volume.VolumePlugin{}
-
-	allPlugins = append(allPlugins, awsebs.ProbeVolumePlugins()...)
-	allPlugins = append(allPlugins, gcepd.ProbeVolumePlugins()...)
+	if !utilfeature.DefaultFeatureGate.Enabled(features.CSIMigration) || !utilfeature.DefaultFeatureGate.Enabled(features.CSIMigrationAWS) {
```

Comment:
could a minor refactor be done so we don't need to duplicate this feature gate checking code? Looks like all these various `Probe` functions have generally the same code and the same set of plugins. Wonder if they could be consolidated for the most part

Reply:
Sure. We can discuss a bit if we want to align this more with https://github.com/kubernetes/kubernetes/pull/80353/files#diff-0967aee5156ab01aec200f53706a7135. I think for now, I can group them under `append*LegacyProviderVolumes` like @BenTheElder did.